src/wazuh_db/wdb_metadata.c

```diff
@@ -174,3 +176,34 @@ int wdb_metadata_get_entry (wdb_t * wdb, const char *key, char *output) {
             return -1;
     }
 }
+
+int wdb_metadata_table_check(wdb_t * wdb, const char * key) {
+    sqlite3_stmt *stmt = NULL;
+    int ret = OS_INVALID;
+
+    if (sqlite3_prepare_v2(wdb->db,
+                            SQL_METADATA_STMT[WDB_STMT_METADATA_TABLE_CHECK],
+                            -1,
+                            &stmt,
+                            NULL) != SQLITE_OK) {
+        merror("DB(%s) sqlite3_prepare_v2(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+        return OS_INVALID;
+    }
+
+    if (sqlite3_bind_text(stmt, 1, key, -1, NULL) != SQLITE_OK) {
```

Comment:
Shouldn't we also finalize the statement here? Just wondering.

Reply:
You are right, `sqlite3_finalize(stmt);` added.
According to [https://www.sqlite.org/c3ref/finalize.html](https://www.sqlite.org/c3ref/finalize.html),  **The application must finalize every prepared statement in order to avoid resource leaks.**